ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/ListSourceExecutor.java

```diff
@@ -475,4 +475,11 @@ private SourceDescriptionWithWarnings(
       this.description = description;
     }
   }
+
+  public static String getFullname(final String name) {
+    if (name.contains("\"") || name.chars().noneMatch(Character::isLowerCase)) {
```

Comment:
source names with `-` also need to be quoted. not having the quotes results in a parsing error rather than a `source not found` error though, so that error might not be that confusing for the user.